synapse/lib/cli.py

```diff
@@ -339,9 +342,20 @@ def runCmdLoop(self):
             # want to ignore the exception.
             pass
 
+        if readline.get_current_history_length() == 0:  # pragma: no cover
+            history_path = s_common.getSynPath('.cmdr_history')
+            # We have to ensure the file exists to use append mode
+            with s_common.genfile(history_path) as fd:
+                pass
+            try:
+                readline.read_history_file(history_path)
+            except IOError:
+                pass
+            atexit.register(readline.append_history_file, 1000, history_path)
```

Comment:
doesn't readline natively support writing stuff back to a history file incrementally rather than all at once at exit?

Reply:
as far as I am aware, it does not. However, we could append each successful cmd to the history file, at the overhead of having to open/close a file for each append.